### PR TITLE
Adds first pass of planned intervention architecture (it will evolve)

### PR DESCRIPTION
**Pull request by sldblog**
## What does this pull request do?

Defines the target architecture for the new dynamic framework digital service.

ℹ️ Everything faded out is `planned` and does not exist yet -- up for debate and review.

⚠️ I arranged this diagram manually. The generated layout has the same nodes but quite a useless layout 😢 I'm working on that on a separate branch.

### Annotations

- `community-api` will have to change
- `intervention-probation-translator` may be done within `community-api` itself; it depends whether we can work together, or need to do push to other services ourselves

### C4 container diagram

structurizr-55488-interventions-container(6)

## What is the intent behind these changes?

To be able to share the existing thinking around what's required for integration and what the first pass will be for the beta phase.

### CODE REVIEW

File: src/main/kotlin/model/Interventions.kt [hunk 1]
```diff
@@ -0,0 +1,144 @@
+package uk.gov.justice.hmpps.architecture
+
+import com.structurizr.model.Container
+import com.structurizr.model.Model
+import com.structurizr.model.Person
```
Comment (mxco86) [inline src/main/kotlin/model/Interventions.kt:5]: This import can be removed as it's unused

File: src/main/kotlin/model/Interventions.kt [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    }
+
+    override fun defineRelationships() {
+      defineAuthentication()
+      defineSharing()
+      defineUsers()
+    }
+
+    fun defineAuthentication() {
+      service.uses(HMPPSAuth.app, "authenticates, authorises users and requests access tokens from", "OAuth2/JWT")
```
Comment (mxco86) [inline src/main/kotlin/model/Interventions.kt:100]: This is not named very well in the HMPPS-Auth class - will push a fix

**Review by mxco86 — COMMENTED** [review]
Tiny comment but other than that it looks good. I'll push a fix for the HMPPS-Auth API naming once this is merged

File: src/main/kotlin/model/Interventions.kt (hunk 1 above)
Comment (sldblog) [reply src/main/kotlin/model/Interventions.kt]: thank you! fixed in c4b79f9ea098d0823ba5433f479ae52f3d835807

**Review by mxco86 — APPROVED** [review]

File: src/main/kotlin/model/Interventions.kt [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+      service.uses(Delius.communityApi, "retrieves list of dynamic framework providers and probation regions from", "REST/HTTP")
+      service.uses(Delius.communityApi, "retrieves current service user appointments and sentence details from", "REST/HTTP")
+      service.uses(Delius.offenderSearch, "searches service user by identity and gets basic identification details from", "REST/HTTP")
+      service.uses(OASys.assessmentsApi, "retrieves service user current risks and needs from", "REST/HTTP")
+
+      translator.uses(Delius.communityApi, "maintains contacts, appointments, registrations with", "REST/HTTP")
+      collector.uses(AnalyticalPlatform.landingBucket, "pushes intervention data daily to")
+    }
+
+    fun defineUsers() {
```
Comment (mxco86) [inline src/main/kotlin/model/Interventions.kt:112]: The users also interact directly with HMPPS-Auth to authenticate and authorise. Not sure if we want to represent this?

Comment (sldblog) [reply src/main/kotlin/model/Interventions.kt:112]: I'm happy either way. What is your preference?

Comment (mxco86) [reply src/main/kotlin/model/Interventions.kt:112]: Showing the direct interaction is more accurate so if it doesn't confuse things too much I'd prefer we did that

Comment (sldblog) [reply src/main/kotlin/model/Interventions.kt:122]: done in 10ddd36

**Pull request merged**